openest/test/generate/test_functions.py

```diff
@@ -1,19 +1,42 @@
 import numpy as np
 import pandas as pd
 import xarray as xr
+import pytest
 
 from openest.generate.base import Constant
 from openest.generate.daily import YearlyDayBins
-from openest.generate.functions import Scale, Instabase, SpanInstabase
+from openest.generate.functions import Scale, Instabase, SpanInstabase, Clip
 from .test_daily import test_curve
 
 
+class MockApplication:
```

Comment:
I think your intention would be clearer if you have this derive from both `Calculation` and `Application`, the way that `CustomFunctionalCalculation` does.

Reply:
I want to keep mocks and stubs as minimal as possible to test just the units I need to test. I think it's okay to inherit from an abstract class but in this case I'm not actually using/implementing the interface suggested by these abstract classes so it's misleading. At the same time I see what you're saying about making my intentions clearer...

In this case I'm going to reject this suggestion but that said, I will likely flesh this out and refactor this as I put in more tests for the other `Calculation` subclasses. So, changes may be quickly coming down the line to address your concern.